operator/cmd/syncclusterconfig/sync.go

```diff
@@ -231,8 +239,28 @@ func mapConvertibleTo[T any](logger logr.Logger, array []any) []T {
 	return converted
 }
 
+type SyncerMode int
+
+const (
+	SyncerModeAdditive = SyncerMode(iota)
+	SyncerModeDeclarative
+)
+
+var ErrSyncerMode = errors.New("unrecognised syncer mode")
+
+func StringToMode(m string) (SyncerMode, error) {
+	switch m {
+	case "additive":
+		return SyncerModeAdditive, nil
+	case "declarative":
+		return SyncerModeDeclarative, nil
+	}
+	return SyncerModeAdditive, ErrSyncerMode
```

Comment:
nit: for clarity on the CLI output: `errors.Wrapf(ErrSyncerMode, "%s", m)`

Reply:
Done